### DISCUSSION TITLE
2008/07/10/%E7%95%8C%E9%9D%A2%E8%AE%BE%E8%AE%A1%E6%8A%80%E5%B7%A7

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#230] [Announcements]
# 2008/07/10/%E7%95%8C%E9%9D%A2%E8%AE%BE%E8%AE%A1%E6%8A%80%E5%B7%A7

最新更新请查看 https://lv.qizhen.xyz

https://blog.qizhen.xyz/2008/07/10/%E7%95%8C%E9%9D%A2%E8%AE%BE%E8%AE%A1%E6%8A%80%E5%B7%A7

**Comment by ruanqizhen** [comment]
之前博客下的留言：

帆
2009年02月25日 12:42 编辑
这么好玩的VI被你们雪藏起来，好无奈啊。

回复
Qizhen
2009年02月25日 09:45 编辑
LabVIEW中只有在函数选板上出现的VI才有文档说明。其它的VI都是不建议用户使用的，所以没有说明。没有文档的VI都是开发者在开发某些LabVIEW功能时创建的子VI，并不打算直接提供给用户使用，所以也没提供文档。NI公司官方是不会对这些VI提供支持和服务的。我知道这个VI是因为这个VI是我写的:)

回复
帆
2009年02月24日 18:28 编辑
实在太谢谢了。我很想知道您是怎么就知道Get Control 这个有意思的VI的。有没有比较系统介绍LabVIEW附带的类似Get Control子VI的文献或者书籍？我以前都不知道这个LV文件夹里有这种好东西。

回复
Qizhen
2009年02月23日 09:40 编辑
帆王:你好。 LabVIEW中有一小部分VI或者函数只能在LabVIEW编程环境下使用，不能脱离LabVIEW在生成的EXE中使用。尤其是一些能够得到VI本身信息，或者改变VI设置的子VI或函数。Get Control.vi就是其中一个。如果想在生成的EXE中得到某个控件的引用还是只能使用最基础的一些方法。

回复
帆
2009年02月21日 21:34 编辑
《黑白棋制作—— LabVIEW 界面设计探索》这篇文章给我不少启发，我觉得其中一个亮点是子VI，Get Control.vi的使用，直到看到这篇文章还不知道LabVIEW自带了那么神奇的VI。（别笑话我，我是个这个寒假刚刚学习LabVIEW的一个大一学生。）不过我在利用Get Control这个子VI的时候，遇到了一个问题。就是用它制作的VI，生成的可执行文件总有问题。就是涉及Get Control的功能时，总是没法实现。调用这个子VI的VI在生成可执行文件时有什么特殊的设置么？希望能帮助指点迷津。谢谢了

回复
立
2008年07月12日 20:50 编辑
期待楼主的专业程序，到时候可以观摩学习下

回复
Qizhen
2008年07月12日 09:18 编辑
嘿嘿，我打算把这个话题写完之后，作出一个非常专业的黑白棋游戏来:)

回复
立
2008年07月11日 21:17 编辑
值得学习！哪天有空做个你文中所描述的例子，也能锻炼下

回复

**Comment by ruanqizhen** [comment]
之前博客下的留言：

Qizhen
2008年08月18日 18:00 编辑
我这里没有 LabVIEW 7.1，所以没法做实验。不过即便不能删除也没关系。LabVIEW 中还有经典风格的控件，是没有边框的。你需要没有边框的自定义控件，可以以经典风格的控件为基础，做修改。

回复
chengzi
2008年08月18日 16:20 编辑
阮兄：你好！
请问是不是LV7.1就不能按照你上面的方法直接删除一个控件的某一部分啊，例如：你上面提到直接删除边框，7.1好像不能直接删除？

回复

**Comment by ruanqizhen** [comment]
之前博客下的留言：

Qizhen
2009年10月11日 11:14 编辑
Pict Ring 只能改变边框的大小，里面的图片是不会跟着调整的。所以一开始就要设计成适合的大小。

回复
嵘
2009年10月11日 09:19 编辑
LED灯做的棋子能用属性节点缩放，Pict Ring 控件是怎么缩放的？俺导入PNG图片后，只能固定大小，控件框缩小后棋子就看不全了。

回复
lihua
2009年04月15日 22:00 编辑
再次表示感谢！没看你相片前，觉得能写出这样的东西的人，应该在40左右或许更年长，看了以后有些吃惊！确实了不起，向你学习！

回复
Qizhen
2009年04月14日 09:04 编辑
隐藏边框需要使用经典样式控件中的pict ring控件。

回复
lihua
2009年04月11日 21:21 编辑
不知道是版本的问题还是我自己没有把握要领，我用的是8.6的，不知道能否说得更详细一些！谢谢！

回复
lihua
2009年04月11日 21:19 编辑
你的这本书，我看了很长时间，也了解到了不少东西，当然很多东西还需要更深的理解和实践，在这里谢谢你！关于上面用pict ring做chess这里，我试着做了一下，不知是我自己没有领会到，还是怎么回事，在隐藏三个图片的时候，却发现最终有一张的底边框怎么也隐藏不了，

回复

**Comment by ruanqizhen** [comment]
之前博客下的留言：

H
2009年07月27日 10:27 编辑
你好，你的文章写得很好，赞一下~~~顺便问一下，那个源码现在可以去哪里下呢？

回复
Qizhen
2008年07月16日 16:09 编辑
我打算写三十万字，可是这样的话，每天写一千字，也还得要300天才行。现在发现，任务太艰巨啦。

回复
zomby
2008年07月15日 22:00 编辑
实现一个功能的方法很多，很多软件工程师都可以找到自己习惯的方法去实现。大侠举一反三地将LabVIEW的多种技术通过一个游戏的例子展示出来，这是很新颖的做法。国内的书籍，都比较生硬死板，可以说就是说明书的翻译。
大侠如果可以顺利出书，一定要买一本。

回复
杰
2008年07月14日 22:08 编辑
软虾最近很高产啊，真的是为出版加油啦，呵呵，支持支持～

回复
